climpred/stats.py

```diff
@@ -319,8 +335,7 @@ def decorrelation_time(da, r=20, dim='time'):
           p.373
 
     """
-    one = da.mean(dim) / da.mean(dim)
-    return one + 2 * xr.concat(
+    return (da.isel(time=0, drop=True) / da.isel(time=0, drop=True)) + 2 * xr.concat(
```

Comment:
To avoid this costly computation you can just do:

`xr.ones_like(da.isel({dim: 0}))`.

Reply:
I tried that but I got ones everywhere and not only where the ocean non-nans are. any idea to do that in a line?

Reply:
Oh that makes sense. I guess you could do:

```python
one = xr.ones_like(da.isel({dim: 0}))
one = one.where(da.isel({dim: 0}).notnull())
```

Might be lower cost and more explicit?